### DISCUSSION TITLE
How to troubleshoot continual disconnects?

### DISCUSSION THREAD

**Discussion by sheldonhull** [community/community#40473] [Codespaces]
I'm trying to rely on Codespaces for the majority of my development.
I have a project that continues to disconnect from both desktop and browser. 

I'm able to get to it but it keeps interrupting me.

I'm on a gigabit internet connection that doesn't seem to have any other apps having failures, and not certain how to troubleshoot what might be causing it. Before I do another rebuild I figured I'd ask in case their are some steps I can take to confirm if it's me or the codespace having issues.

- 4 core machine config (32 gb ram)
- Enterprise account for this one
- 7-30% cpu usage
- 3.3 of 7.77gb used. `/dev/sdb1 36% used, /dev/loop0 58% used` (from resource monitor extension). 

```text
           vscode@codespaces-xxxxxxx 
        `:+ssssssssssssssssss+:`           ------------------------ 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 22.04.1 LTS x86_64 
    .ossssssssssssssssssdMMMNysssso.       Host: Virtual Machine 7.0 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Kernel: 5.4.0-1094-azure 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Uptime: 48 mins 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Packages: 449 (dpkg) 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Shell: bash 5.1.16 (👉🏻 i'm using zsh)
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Terminal: vscode 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   CPU: Intel Xeon Platinum 8272CL (4) @ 2.593GHz 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   GPU: 00:08.0 Microsoft Corporation Hyper-V virtual VGA 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Memory: 3102MiB / 7957MiB 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   CPU Usage: 10% 
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Disk (/): 19G / 32G (62%) 
  +sssssssssdmydMMMMMMMMddddyssssssss+

```

**Comment by cdwort** [comment]
Hello @sheldonhull, thanks for your report! It sounds like this is happening across multiple codespaces for you? Could you share the codespace URL from the browser (this has the codespace name as the subdomain, which I'll use to take a look)?

**Reply by sheldonhull** [reply]
https://sheldonhull-stunning-space-guacamole-7pp7x5r4962rxj6.github.dev/


Mmmmm. Space guacamole. 😆

**Reply by cdwort** [reply]
😂 Those space avocados really are out of this world!

Can you describe what it happening/what you are experiencing on this (and other?) codespaces? 

I can see that you started this codespace about an hour ago and then about 3 minutes after posting you hard stopped and restarted the codespace. I do see something interesting at 18:20 UTC, so about 25 minutes into you using the codespace. Does that align with what you remember?

**Reply by sheldonhull** [reply]
Yes. I had random disconnects and was unable to use from desktop. I had slightly better result from browser. This repeated last night and today.

**Reply by cdwort** [reply]
Thanks for that. The next step is taking a look at the logs. Would you be willing to [open a support ticket](https://support.github.com/contact/bug-report) with the codespace logs that you export _after_ you're disconnected and manage to reconnect? Here are instructions for how to get to them on [desktop](https://docs.github.com/en/codespaces/troubleshooting/github-codespaces-logs?tool=vscode#codespace-logs-1) and [web](https://docs.github.com/en/codespaces/troubleshooting/github-codespaces-logs?tool=webui#codespace-logs). You could upload them here, but I don't think I can guarantee that everything in there is something you would want to be public.

**Reply by sheldonhull** [reply]
Yes I'll do that. Thanks for prompt support. Great experience. Was concerned using discussions it was the wrong route but good experience. 

In future do I do that log submission and bypass this if it's a similar complicated debugging issue or need both?

**Reply by cdwort** [reply]
I'd definitely say there's no harm in _checking_ the discussions for something that looks like your issue, but if you can tell it's likely going to be a somewhat unique issue, go ahead and create a support ticket first.
